lib/parser.js

```diff
@@ -28,30 +28,38 @@ function parser (data, opts) {
 
     while (buffer.length) {
       if (this.readingHeader) {
-        readAndConsume(buffer, this.headers, 8 - this.headers.length)
+        var readHeaders = buffer.shallowSlice(0, 8 - this.headers.length)
+
+        this.headers.append(readHeaders)
+
+        buffer = buffer.shallowSlice(readHeaders.length)
 
         if (this.headers.length !== 8) {
           return cb()
         }
 
         this.readingHeader = false
         this.payloadLength = this.headers.readUInt32BE(4)
-        this.headers.consume(8)
+        this.headers = bl()
```

Comment:
I thought it would look a bit more explicit, but don't mind changing it back, I will change back line 71 as well, thanks